### ISSUE TITLE
[Feature Request] Auto tiling

### ISSUE THREAD

**Issue by Vixeliz** [StarArawn/bevy_ecs_tilemap#389]
I think this would be a very useful feature for some projects. Something like the ldtk rules systems allows really powerful procedural map generation. And also can be used for example letting players edit tiles without having to manually select the tile sprites needed. Is this something you may be interested in adding? Awesome library Ive really enjoyed using it!

**Comment by frederickjjoubert** [comment]
I tried implementing my own auto tiling system on top of `bevy_ecs_tilemap` you can check out the example I made [here](https://github.com/frederickjjoubert/bevy_ecs_tilemap_autotile_example) if you're interested.

My code is very verbose, but it works. Best of luck.

**Comment by Vixeliz** [comment]
> I tried implementing my own auto tiling system on top of `bevy_ecs_tilemap` you can check out the example I made [here](https://github.com/frederickjjoubert/bevy_ecs_tilemap_autotile_example) if you're interested.
> 
> My code is very verbose, but it works. Best of luck.

Nice after I posted this I had actually started working on making a library for a lib for autotiling I found called “bevy_tileset”

**Comment by StarArawn** [comment]
After experimenting with auto tiling, and the vast number of ways you can achieve auto tiling, I'm not sure I'm convinced a generalized solution is something that we want to add to `bevy_ecs_tilemap`. My thoughts are that auto tiling features should probably live in a separate crate. Is there a specific reason it should be a part of this crate?

**Comment by Vixeliz** [comment]
Firstly I am no longer in need of this so I can’t personally speak. However looking at gamedev as a whole autotiling in 2d tile based games is super common and having solution that’s easy to use built in would probably be nice for a lot of people but of course is something that could live in a separate crate. Either way as said I am no longer using this crate feel free to leave open if you want to have further discussion or if you are uninterested feel free to close.

**Comment by LemonjamesD** [comment]
> Firstly I am no longer in need of this so I can’t personally speak. However looking at gamedev as a whole autotiling in 2d tile based games is super common and having solution that’s easy to use built in would probably be nice for a lot of people but of course is something that could live in a separate crate. Either way as said I am no longer using this crate feel free to leave open if you want to have further discussion or if you are uninterested feel free to close.

Yeah I am in need of this for my game that I am working on and I don't really want to roll my own as I don't really want to learn how and don't know how to start at all.

**Comment by griffenliu** [comment]
People like things out of the box, like godot's tilemap component.
